server/expression/comparison.go

```diff
@@ -4,10 +4,17 @@ import (
 	"fmt"
 
 	"github.com/kubeshop/tracetest/server/assertions/comparator"
+	"github.com/kubeshop/tracetest/server/expression/types"
 	"github.com/kubeshop/tracetest/server/expression/value"
 )
 
 func compare(comparatorName string, leftValue, rightValue value.Value) error {
+	if rightValue.Value().Type == types.TypeJson && comparatorName == "contains" {
+		if leftValue.Value().Type == types.TypeJson || leftValue.Value().Type == types.TypeArray {
+			comparatorName = "json-contains"
+		}
+	}
```

Comment:
I thought this was cleaner to write than a longer `if` statement containing all four conditions.